In [11]:
import math

# Calculating 1000 factorial
result = math.factorial(1000)
print(f"1000 factorial = {result}")

# Counting trailing zeros
def count_trailing_zeros(n):
    count = 0
    power_of_5 = 5
    while power_of_5 <= n:
        count += n // power_of_5
        power_of_5 *= 5
    return count

zeros = count_trailing_zeros(1000)
print(f"Number of trailing zeros in 1000 factorial: {zeros}")

# Verifying by converting to string and counting zeros from the end
result_str = str(result)
trailing_zeros_count = 0
for i in range(len(result_str) - 1, -1, -1):
    if result_str[i] == '0':
        trailing_zeros_count += 1
    else:
        break

print(f"Verified by counting: {trailing_zeros_count} trailing zeros")

1000 factorial = 40238726007709377354370243392300398571937486421071463254379991042993851239862902059204420848696940480047998861019719605863166687299480855890132382966994459099742450408707375991882362772718873251977950595099527612087497546249704360141827809464649629105639388743788648733711918104582578364784997701247663288983595573543251318532395846307555740911426241747434934755342864657661166779739666882029120737914385371958824980812686783837455973174613608537953452422158659320192809087829730843139284440328123155861103697680135730421616874760967587134831202547858932076716913244842623613141250878020800026168315102734182797770478463586817016436502415369139828126481021309276124489635992870511496497541990934222156683257208082133318611681155361583654698404670897560290095053761647584772842188967964624494516076535340819890138544248798495995331910172335555660213945039973628075013783761530712776192684903435262520001588853514733161170210396817592151090778801939317811419454525722386554146106289218

Ans no 1.
To find how many zeros 1000! ends with, I need to count the number of times 10 divides into 1000!. Since 10 = 2 × 5, and there are always more factors of 2 than 5 in a factorial, I need to count the factors of 5.
The formula is: ⌊n/5⌋ + ⌊n/25⌋ + ⌊n/125⌋ + ⌊n/625⌋ + ...
For 1000!:

  ⌊1000/5⌋ = 200
 ⌊1000/25⌋ = 40
  ⌊1000/125⌋ = 8
  ⌊1000/625⌋ = 1
   ⌊1000/3125⌋ = 0

Total: 200 + 40 + 8 + 1 = 249, 


In [7]:
# Calculating 100^13 - 26
number = 100**13 - 26
print(f"100^13 - 26 = {number}")

# Sum of digits
digit_sum = sum(int(digit) for digit in str(number))
print(f"Sum of digits: {digit_sum}")

# Verifying the calculation
print(f"Manual check: 24×9 + 7 + 4 = {24*9 + 7 + 4}")


100^13 - 26 = 99999999999999999999999974
Sum of digits: 227
Manual check: 24×9 + 7 + 4 = 227


Ans no 2.
 It's 100^13 - 26,
Let me calculate this correctly:
100^13 = (10^2)^13 = 10^26
   10^26 = 1 followed by 26 zeros = 100000000000000000000000000
   100^13 - 26 = 100,000,000,000,000,000,000,000,000 - 26
     = 99,999,999,999,999,999,999,999,974
This number has:
Twenty-four 9's
One 7 One 4

Sum of digits = 24 × 9 + 7 + 4 = 216 + 7 + 4 = 227
Answer: 227

In [5]:
# Problem 3: Soccer Teams Handshakes
between_teams = 11 * 11
within_team = 11 * 10 // 2
total = between_teams + within_team
print(f"Total handshakes: {total}")

Total handshakes: 176


Ans no 3 
Handshakes:
Each team has 11 players, so there are 22 people total.

      Between the two teams: 11 × 11 = 121 handshakes
Within the winning team: C(11,2) = 11×10/2 = 55 handshakes (1st player will do 10, 2nd will do 8.... and it goes on till 1)
Total: 121 + 55 = 176 handshakes



In [ ]:
# Problem 4
Total people
people = 10  

# Possible handshakes for one person (0 to 8)
handshakes = list(range(9))  

# One number must be left for Mr. Smith
mr_smith = 4  

print("Handshakes reported by guests:", handshakes)
print("Mr. Smith shook", mr_smith, "hands")


In [ ]:
Harder Handshakes:
There are 10 people total (5 couples).
      Since Mr. Smith received a different answer from each person, and there are 9 other people, the possible answers can be 0, 1, 2, 3, 4, 5, 6, 7, 8.
 No one shakes hands with their spouse, the maximum anyone can shake is 8 hands.
If someone shook 8 hands, their spouse shook 0 hands (since they can't shake the same people).
If someone shook 7 hands, their spouse shook 1 hand.
And so on...
The pairs are: (0,8), (1,7), (2,6), (3,5), and one person with 4 handshakes.
Since Mrs. Smith is the only one unaccounted for, she shook 4 hands.
Answer: Mr. Smith shook hands 4 times.